src/py42/sdk/__init__.py

```diff
@@ -44,77 +76,74 @@ def from_local_account(cls, host_address, username, password):
         Returns:
             :class:`py42.sdk.SDKClient`
         """
-        session_impl = Session
-        session_factory = SessionFactory(
-            session_impl, SessionModifierFactory(), AuthHandlerFactory()
-        )
-        basic_auth_session = session_factory.create_basic_auth_session(
-            host_address, username, password
-        )
-        sdk_dependencies = SDKDependencies(
-            host_address, session_factory, basic_auth_session
-        )
-        return cls(sdk_dependencies)
+        auth = None
+        if username and password:
+            auth = HTTPBasicAuth(username, password)
+        auth_connection = Connection.from_host_address(host_address, auth=auth)
+        auth = V3Auth(auth_connection)
```

Comment:
Aren't you overriding line 81 here?

Reply:
I see it gets uses, just sharing a variable is confusing

Reply:
def can change this to be less confusing

Reply:
worth discussing for sure, but as part of this refactor i think we should just aim to preserve existing behavior